Build failure: mdbtools (darwin-aarch64)

### Steps To Reproduce

Steps to reproduce the behavior:
1. build mdbtools (e.g. `nix run nixpkgs#mdbtools`)

### Build log

```
make[3]: Entering directory '/private/tmp/nix-build-mdbtools-1.0.0.drv-0/source/src/sql'
  CC       mdbsql.lo
  CC       parser.lo
  LEX      lexer.c
  CC       lexer.lo
parser.c:1053:9: error: variable 'yynerrs' set but not used [-Werror,-Wunused-but-set-variable]
    int yynerrs = 0;
        ^
1 error generated.
```

### Additional context

I've fixed this locally by setting `CFLAGS = -Wno-error`:

```nix
pkgs.mdbtools.overrideAttrs {
  NIX_CFLAGS_COMPILE = "-Wno-error";
}
``` 

### Notify maintainers

@r-ryantm (not listed as maintained, but did the previous version bump)

### Metadata

```console
$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"aarch64-darwin"`
 - host os: `Darwin 23.1.0, macOS 14.1.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.18.1`
 - nixpkgs: `/etc/nix/path/nixpkgs`
```

### Priorities

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/tools/misc/mdbtools/default.nix
+++ b/pkgs/tools/misc/mdbtools/default.nix
@@ -16,6 +16,8 @@ stdenv.mkDerivation rec {
 
   configureFlags = [ "--disable-scrollkeeper" ];
 
+  env.NIX_CFLAGS_COMPILE = lib.optionalString stdenv.cc.isClang "-Wno-error=unused-but-set-variable";
+
   nativeBuildInputs = [
     pkg-config bison flex autoreconfHook txt2man which
   ];

